src/preprocess/reduction/pca.py

```diff
@@ -0,0 +1,56 @@
+from state import State
+import numpy as np
+import pandas as pd
+
+
+class PCAReducer:
+    def __init__(self, state: State, acceptable_ratio = 0.05):
+        self.state = state
+        self.acceptable_ratio = acceptable_ratio
+
+    def reduce(self, dim_number):
```

Comment:
```suggestion
    def reduce(self, dim_number: int | None) -> None:
```
I think we would benefit from type annotating public methods like this one

Reply:
Or you can write:
```suggestion
    def reduce(self, dim_number: Optional[int]) -> None:
```